doc/_static/js/toggle_diffs.js

```diff
@@ -0,0 +1,68 @@
+// All diffs will have one of the below classes
+const changeClasses = ["change-replaced", "change-replacement", "change-added", "change-removed"];
+// The list of components to be scanned for diffs - selected by ID
+const componentsList = [
+    '#module-details > *',
+    '#module-contents > *',
+    '#package-details > *',
+    '#package-contents > *', 
+    '#subpackages',
+    '#submodules',
+    '#table-of-contents',
+    '#introduction'
+];
+const components = componentsList.join(", ")
+
+function toggleElements() {
+
+    // Check if the entire page is replaced. If yes, do nothing
+    var fullPageReplaced = false;
+    const pageElement = document.querySelector('span.target');
+    if (changeClasses.some(r => pageElement.classList.value.includes(r))){
+        fullPageReplaced = true;
+    }
+
+    // Do nothing if the full page is replaced
+    if (!fullPageReplaced){
+        // Examine the contents of the individual sections
+        // Hide sections that do not have diffs in any of the children
+        const all = document.querySelectorAll(components);
+
+        for (i=0; i < all.length; i++){
+    
+            const hasDiffs = childrenHaveDiffs(all[i]);
+            if (!hasDiffs){
+                all[i].classList.toggle("hideElement");
+            };
+        }
+    }
+
+    // Toggle button text depending on what is currently displayed
+    const btn = document.querySelector('.diff-toggle');
+    if (btn.innerHTML === "Only Show Diffs") {
+        btn.innerHTML = "Show All";
+    } else {
+        btn.innerHTML = "Only Show Diffs";
+    }
+}
+
+function childrenHaveDiffs(element){
+    
+    
+    var hasDiffs = false;
+
+    if (changeClasses.some(r => element.classList.value.includes(r))){
+        hasDiffs = true;
+    } else {
+        const children = element.querySelectorAll('*');
+
+        for (j = 0; j < children.length; j++){
+            if (changeClasses.some(r => children[j].classList.value.includes(r))){
+                hasDiffs = true;
+                break;
+            } 
+        }
+    }
+
+    return hasDiffs;
```

Comment:
Perhaps:

```suggestion
    if (changeClasses.some(r => element.classList.value.includes(r))){
        return true;
    } else {
        const children = element.querySelectorAll('*');

        for (j = 0; j < children.length; j++){
            if (changeClasses.some(r => children[j].classList.value.includes(r))){
                return true;
            } 
        }
    }

    return false;
```

Also, would it be better to join the classes with `, ` like you do for `componentsList` and let the browser handle it?

Reply:
Will update the `childrenHaveDiffs` function. 
Sorry. I didn't quite fully get the comment about joining the classes with `, `.

Reply:
Instead of:

```js
const children = element.querySelectorAll('*');
```

How about:

```js
const children = element.querySelectorAll("." + changeClasses.join(", ."));
```

I totally could be way off the mark about what this does though ^^;;

Reply:
Good idea. That way I avoid the for loop. Updated